lib/ansible/modules/cloud/online/online_server_install.py

```diff
@@ -0,0 +1,137 @@
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: online_server_install
+short_description: Install an Online server
+description:
+  - Gather facts about the servers.
+  - U(https://www.online.net/en/dedicated-server)
+version_added: "2.8"
+author:
+  - "Ludovic Petetin (@broferek)"
+extends_documentation_fragment: online
+'''
+
+EXAMPLES = r'''
+- name: Install Centos on the server 5824 with the user foo created
+  online_server_install:
+    api_token: '0d1627e8-bbf0-44c5-a46f-5c4d3aef033f'
+    host: "5824"
+    os_id: 305
+    hostname: "sd-5824"
+    root_password: "root_bar"
+    user_login: "foo"
+    user_password: "bar"
+    partitioning_template_ref: "a645ex65-af43-4rr8-bc45-654qsdf654e2"
+'''
+
+RETURN = r'''
+---
+msg:
+    description: Message specifying that the installation has started
+    returned: success
+    type: string
+    sample: The installation has started. Check your server state at https://api.online.net/fr/server/state/5824
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.online import (
+    Online, OnlineException, online_argument_spec
+)
+
+
+class OnlineServerInstall(Online):
+
+    def __init__(self, module):
+        super(OnlineServerInstall, self).__init__(module)
+        host = module.params['host']
+        self.name = 'api/v1/server/' + host
+
+
+def main():
+    argument_spec = online_argument_spec()
+    argument_spec.update(
+        dict(
+            host = dict(
+                required = True,
+                aliases = ['server']
+            ),
+            os_id = dict(
+                required = True,
+                type = 'int'
+            ),
+            hostname = dict(
+                required = True
+            ),            
+            root_password = dict(
+                required = True,
+                no_log = True
+            ),            
+            user_login = dict(
+                required = True
+            ),            
+            user_password = dict(
+                required = True,
+                no_log = True
+            ),            
+            partitioning_template_ref = dict(
+                required = True
+            )            
+        )    
+    )
+             
+    module = AnsibleModule(
+        argument_spec = argument_spec,
+        supports_check_mode = True
+    )
+    host = module.params['host']
+    post_data = dict(        
+        os_id = module.params['os_id'],
+        hostname = module.params['hostname'],
+        root_password = module.params['root_password'],
+        user_login = module.params['user_login'],
+        user_password = module.params['user_password'],
+        partitioning_template_ref = module.params['partitioning_template_ref']
+    )
+    try:
```

Comment:
Could you have a shorter try block? I cannot see what function is supposed to raise an exception here.

Reply:
This is actually for both API calls (get_resources and post_resources), I will fix the try catch around these 2 lines only